File: 2025-07-July-GFG-POTD/July 02 - Longest subarray with Atmost two distinct integers/Solution.cpp
#include <unordered_map>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int totalElements(vector<int>& arr) {
        unordered_map<int, int> map;
        int maxLen = 0, start = 0;

        for (int end = 0; end < arr.size(); end++) {
            map[arr[end]]++;

            while (map.size() > 2) {
                map[arr[start]]--;
                if (map[arr[start]] == 0) {
                    map.erase(arr[start]);
                }
                start++;
            }

            maxLen = max(maxLen, end - start + 1);
        }

        return maxLen;
    }
};

File: 2025-07-July-GFG-POTD/July 02 - Longest subarray with Atmost two distinct integers/Solution.py
from collections import defaultdict

class Solution:
    def totalElements(self, arr):
        count = defaultdict(int)
        max_len = start = 0

        for end in range(len(arr)):
            count[arr[end]] += 1

            while len(count) > 2:
                count[arr[start]] -= 1
                if count[arr[start]] == 0:
                    del count[arr[start]]
                start += 1

            max_len = max(max_len, end - start + 1)

        return max_len
